console-subscriber/src/lib.rs

```diff
@@ -789,87 +789,59 @@ where
     }
 
     fn on_enter(&self, id: &span::Id, cx: Context<'_, S>) {
-        fn update<S: Subscriber + for<'a> LookupSpan<'a>>(
-            span: &SpanRef<S>,
-            at: Option<Instant>,
-        ) -> Option<Instant> {
+        if let Some(span) = cx.span(id) {
+            let now = Instant::now();
             let exts = span.extensions();
             // if the span we are entering is a task or async op, record the
             // poll stats.
             if let Some(stats) = exts.get::<Arc<stats::TaskStats>>() {
-                let at = at.unwrap_or_else(Instant::now);
-                stats.start_poll(at);
-                Some(at)
+                stats.start_poll(now);
             } else if let Some(stats) = exts.get::<Arc<stats::AsyncOpStats>>() {
-                let at = at.unwrap_or_else(Instant::now);
-                stats.start_poll(at);
-                Some(at)
+                stats.start_poll(now);
             // otherwise, is the span a resource? in that case, we also want
             // to enter it, although we don't care about recording poll
             // stats.
             } else if exts.get::<Arc<stats::ResourceStats>>().is_some() {
```

Comment:
I think it might make more sense to put this comment inside the (now) empty block following the `else if`

```suggestion
            } else if exts.get::<Arc<stats::ResourceStats>>().is_some() {
                // otherwise, is the span a resource? in that case, we also want
                // to enter it, although we don't care about recording poll
                // stats.
```